docs/devops.md

```diff
@@ -10,7 +10,334 @@ The stack is managed using the following tools:
+            - IF NOT MIGRATING: Remove the lines relating to Fedora.
+        
+        - Edit `saml.yaml`: [or don't?]
+            - "contains the configuration for SimpleSAMLphp and needs to be configured per site to use saml login. Due to the required AWS permissions the values for the dgi sso will need to be provided by dgi.
+            - Lets us use DGI's sso on the site. Gives us admin access using the same account across sites, and can allow dgi staff to access the site.
+            - We must provide DGI with a list of site URLs ahead of time.
+            - DGI will provide a `saml.yaml` file for each site.
+            - If you want to have sso just for your own account you could create a free tier Jumpcloud account and manage the sso yourself.
+    - `secrets` directory: `cd /opt/helm_values/[siteName]/secrets`
+        - Edit `values.yaml`:
+            - Set `secretPrefix` to your new site's name in the pattern `bceln/siteName`
+                - e.g. if you're creating the Okanagan College site, set it to `bceln/oc` 
+                    - `siteName` is the name of the directory you are building your new site in. 
+    - Set the namespace: `/opt/helm_values/[newsite]/extras.yaml`
+        - In the top section:
+            ```
+            apiVersion: v1
+            kind: Namespace
+            metadata:
+              labels:
+                dgicloud.com/drupal.site: [newsite]
+              name: [newsite]
+            ```
+5. Secret generation:
+    - On your personal computer (*not on the server*), run the `gen-secrets.sh` script:
+        - Make sure `jq` is installed: `brew install jq`
+        - Download from Github: `https://github.com/discoverygarden/helm-charts/blob/main/charts/aws-secrets/gen-secrets.sh`
+            - Put it in `~/arca-migration/gen-secrets.sh` for simplicity.
+        - Get authorized with AWS SSO:
+            - aws sso login --profile bceln-deploy
+        - Run `AWS_PROFILE=bceln-deploy ./gen-secrets.sh bceln/[siteName]`
+            - The `AWS_PROFILE` piece identifies which profile is being used when running the script.
+6. Install the site:
+    - In `arca-dc`, go back to the `helm_values` directory: `cd /opt/helm_values`
+    - Run the install script: `./scripts/update-all.sh siteName` where `siteName` is [the name of the directory you created???]
+
+
+# Upgrading a site
+
+Running `update-helm.sh` and `update-all.sh` will deploy the latest release to
+a service. However, some more care is required when updating drupal.
+
+Before updating, drupal configuration should be exported and merged back in.
+The proccess involves checking out the currently deployed tag locally and
+replacing the drupal configs with what is running in production, then creating
+a pull request.
+
+Example:
+```bash
+ssh beln-arca-dc.dc.sfu.ca
+cd /opt/helm_values
+# export the configs
+./scripts/export-config.sh dc
+# get the currently deployed tag
+./scripts/get-tag.sh dc
+exit
+
+# cd to where the repo exists locally
```

Comment:
Not exactly clear here. Are these instructions saying:

1. On your local machine, enter the `bceln-drupal` directory linked to the git repository
2. Checkout a new branch names for the tag (e.g. if the tag is 1.51.2, the branch name  is `dc-reconcile v1.51.2`)
3. `rm -rf config`
4. `scp beln-arca-dc.dc.sfu.ca:/opt/helm_values/dc/config.tar.gz .`
5. `rm config.tar.gz`
6. Confirm everything went into the `config` directory

Does this have to be done as a separate branch for each of our eventual 30+ sites, then?

Reply:
@Alexander-Cairns The above question is for you. Will your script take care of the majority of this, or is there still going to be a lot of manual git work going forward?

Reply:
#63 Adds the script and documents how to use it. It can now all be run from the server. We are resolving an issue with the splits so you should hold off on running it untill we've resolved the issue.